### PR TITLE
[Infrastructure] windows wrapper for the 'compose.py'

### PR DESCRIPTION
**Pull request by vetall812**
Tool for the windows artists that wraps default multiplatform compose.py

#### Summary
Infrastructure "Windows wrapper for compose.py"

#### Content of the change
This batch script allows artists to set up variables and just doubleclick this file to get composed tileset and put it in game. So it can be tested on the fly.

There are also some checks and useful tips that helps setting artists environment up.

#### Testing
Default run if everything is ok:
image
Quiet mode:
image

Some error handling:
image

#### Additional information
It will also try to install pyvips once in case of installed python but missing module.

### CODE REVIEW

File: tools/updtset.cmd [hunk 1]
```diff
@@ -0,0 +1,235 @@
+@echo off
+setlocal EnableDelayedExpansion
+setlocal EnableExtensions
+:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::ver 1.13 / 2022-06-29::::
+:: this script does following:
```
Comment (Fris0uman) [inline tools/updtset.cmd:5]:
```suggestion
:: this script does the following:
```

File: tools/updtset.cmd [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+:: - compose tileset from the source
```
Comment (Fris0uman) [inline tools/updtset.cmd:6]:
```suggestion
:: - compose tilesets from source
```

File: tools/updtset.cmd [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+:: - put composed tileset into some folder for future use
+:: - copy said composed tileset from the folder into the game so you can try in on the fly
+:: - copy layering json file from tileset source into the game
+::
+:: PLEASE setup 5 variables below, put correct text inside a quotes
```
Comment (Fris0uman) [inline tools/updtset.cmd:11]:
```suggestion
:: PLEASE setup 5 variables below, put correct text between quotes
```

File: tools/updtset.cmd [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
+:: 1. Set correct folder same as in github desktop repository local path
```
Comment (Fris0uman) [inline tools/updtset.cmd:15]:
```suggestion
:: 1. Set path to your local copy of the tileset repository
```

File: tools/updtset.cmd [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+
+::
+:: 2. Set correct tileset folder name (e.g. UltimateCataclysm)
+
+set def_tileset="UltimateCataclysm"
+
+::
+:: 3. Set path to compose.py file 
+::    if it is in cdda source local repository type full path to it
+::    or if it's in your workshop type this path
```
Comment (Fris0uman) [inline tools/updtset.cmd:27]:
```suggestion
```
This part is confusing, no matter where compose.py is you set the path to it, no need to describe special cases

File: tools/updtset.cmd [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+
+set script_dir="T:\workshop"
+
+::
+:: 4. Set path to the folder where to put composed tileset (for future examination)
```
Comment (Fris0uman) [inline tools/updtset.cmd:32]:
```suggestion
:: 4. Set path to the folder where to put the composed tileset (for future examination)
```
I'm not sure what's the point of this option

File: tools/updtset.cmd [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+
+set composed_dir="T:\workshop\compiled"
+
+::
+:: 5. Set path to the folder where the CDDA game located
```
Comment (Fris0uman) [inline tools/updtset.cmd:37]:
```suggestion
:: 5. Set path to CDDA game folder
```

File: tools/updtset.cmd [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+:: + Added installation of pyvips
+:: 1.11
+:: + Added support of paths with spaces
+:: ~ Changed behavior, now verbose is default mode
+:: ~ Changed /v key to /q - quiet mode
+:: 1.10
+:: + Now supports /v /d /s keys
+:: + Also can accept tileset name as argument so it can be provided via shortcut
+:: 1.01
+:: + Added manual reference for python (module+library)
```
Comment (Fris0uman) [inline tools/updtset.cmd:235]:
```suggestion
```
I don't think you need to track changelog in there, it will be tracked by github history

Comment (vetall812) [reply tools/updtset.cmd:233]: that was ages ago )

**Pull request merged**